stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/viewmodel/channel/ChannelListViewModel.kt

```diff
@@ -124,32 +125,50 @@ public class ChannelListViewModel(
      */
     init {
         viewModelScope.launch {
-            searchQuery.combine(queryConfig) { query, config -> query to config }
-                .collectLatest { (query, config) ->
-                    val result = chatDomain.queryChannels(
-                        filter = createQueryChannelsFilter(config.filters, query),
-                        sort = config.querySort
-                    ).await()
-
-                    if (result.isSuccess) {
-                        observeChannels(controller = result.data(), searchQuery = query)
+            if (ToggleService.isEnabled(ToggleService.TOGGLE_KEY_OFFLINE)) {
+                initWithOfflinePlugin()
+            } else {
+                initWithChatDomain()
+            }
+        }
+    }
+
+    private suspend fun initWithChatDomain() {
+        searchQuery.combine(queryConfig) { query, config -> query to config }
+            .collectLatest { (query, config) ->
+                val result = chatDomain.queryChannels(
+                    filter = createQueryChannelsFilter(config.filters, query),
+                    sort = config.querySort
+                ).await()
+
+                if (result.isSuccess) {
+                    observeChannels(controller = result.data(), searchQuery = query)
+                } else {
+                    result.error().cause?.printStackTrace()
+                    channelsState = channelsState.copy(isLoading = false, channelItems = emptyList())
+                }
+            }
+    }
+
+    private suspend fun initWithOfflinePlugin() {
+        searchQuery.combine(queryConfig) { query, config -> query to config }
+            .collectLatest { (query, config) ->
+                val queryChannelsRequest = QueryChannelsRequest(
+                    filter = createQueryChannelsFilter(config.filters, query),
+                    querySort = config.querySort,
+                    limit = CHANNEL_LIMIT,
+                    messageLimit = MESSAGE_LIMIT,
+                    memberLimit = MEMBER_LIMIT,
+                )
+                queryChannelsState = chatClient.asReferenced().queryChannels(queryChannelsRequest).asState(viewModelScope)
+                queryChannelsState?.let {
+                    if (queryChannelsState?.channels?.value?.isNotEmpty() == true) {
```

Comment:
This doesn't work - in case the channels are empty the user just sees an empty screen and nothing happens.

It should be:

```
                queryChannelsState?.let {
                    observeChannels(it, searchQuery = query)
                }
```

And then the channels will just update.